### PR TITLE
Rework mod launchers to address issues with icons on Windows

### LINKED ISSUE #14174: No icon for OpenRA app in Windows taskbar
Issue is reproduced on both Windows 10 and Windows 8.1
unknown
image

However, as @pchote stated, it does show in his Windows configuration.
f4ote30

### PR DESCRIPTION
**Pull request by penev92**
Until now the mod-specific launchers (with a specific name and icon) launched the mod-agnostic `OpenRA.Game.exe` and told it to pretend like it cares what mod it's running. That resulted in some issues with icons at least on Windows.

This changes the launchers to run the game loop directly, so instead of the launcher launching `OpenRA.Game.exe`, it now launches a child process of the same launcher, but in a different state, which runs the game loop. The point of having a parent and child processes is to maintain error handling if the child blows up.

Unfortunately not straight-forward to test on Windows, and apparently creating the launchers doesn't happen on Windows, so will need someone who can create those to provide reviewers with samples.

Fixes #14174.

### CODE REVIEW

**Comment by pchote** [comment]
The two main problems prompting this change in approach are:
* The current use of `SendMessage` works inconsistently, which causes #14174.
* Pinning a running game to the taskbar leaves a broken launcher that can't launch the game.

File: packaging/windows/WindowsLauncher.cs.in [hunk 1]
```diff
@@ -12,104 +12,81 @@
...
-			// Native type: TCHAR[MAX_PATH]
-			[MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-
-			public string szDisplayName;
-
-			// Native type: TCHAR[80]
-			[MarshalAs(UnmanagedType.ByValTStr, SizeConst = 80)]
+			if (args.Any(x => x.Contains("Engine.LaunchPath")))
+			{
+				var exitCode = RunGame(args); // TODO: Test various exit scenarios.
```
Comment (pchote) [inline packaging/windows/WindowsLauncher.cs.in]: This will need to be finished before merging.

Comment (penev92) [reply packaging/windows/WindowsLauncher.cs.in]: Actually upon further inspection I should already have all the error handling covered inside `RunGame`, so there is likely no point in even returning an exit code here.

**Review by pchote — COMMENTED** [review]
Looks good as a first cut on the feature.
[Link to test build](https://www.dropbox.com/s/3np74radbl703d3/OpenRA-packagetest-20171127.exe?dl=0)

I can confirm that this fixes pinning to taskbar, and that the icons work as expected.  The crash dialog itself does not work, however.

A few minor code nits:

File: packaging/windows/WindowsLauncher.cs.in [hunk 2]
```diff
@@ -12,104 +12,75 @@
...
-			var launcherPath = Assembly.GetExecutingAssembly().Location;
-			var directory = Path.GetDirectoryName(launcherPath);
-			var enginePath = Path.Combine(directory, "OpenRA.Game.exe");
+			if (args.Any(x => x.Contains("Engine.LaunchPath")))
+				return RunGame(args);
+
+			return RunInnerLauncher();
+		}
 
-			Directory.SetCurrentDirectory(directory);
```
Comment (pchote) [inline packaging/windows/WindowsLauncher.cs.in]: Why did you remove setting the working directory to the game dir?

**Comment by pchote** [comment]
Updated test build: https://www.dropbox.com/s/cbgp6kk9nqmheo8/OpenRA-packagetest-20171129.exe?dl=0

**Comment by netnazgul** [comment]
unknown
seems fixed

**Comment by reaperrr** [comment]
Woah, return of the "View logs" button! 👍

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/bab3948bc59242e837832700988de8b5a0e8081a...9d207e78f9c52c3c52751ae49fcb7fb925643c00
